Спринт 6

ASevastyanov opened the pull request:
- Добавлена работа с сетью, список фильмов загружается из API IMDb.
 - Обработаны состояния приложения при загрузке фильмов и получения ошибок.

ASevastyanov committed:
Спринт 6
 - Добавлена работа с сетью, список фильмов загружается из API IMDb.
 - Обработаны состояния приложения при загрузке фильмов и получения ошибок.

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -23,10 +23,11 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
+    @IBOutlet private weak var textLabel: UILabel!
+    @IBOutlet private weak var counterLabel: UILabel!
+    @IBOutlet private weak var yesButtonClicked: UIButton!
+    @IBOutlet private weak var noButtonClicked: UIButton!
```
Использование имен `yesButtonClicked` или `noButtonClicked` может вызвать путаницу, потому что эти имена похожи на названия функций или каких-то действий, хотя это просто кнопки. Рекомендую переименовать их.
Например, имена `yesAnswerButton` и `noAnswerButton` явно указывают, что это объекты кнопок, связанные с ответами 'да' и 'нет' в  квизе. Это сделает код более понятным и читаемым, особенно для других разработчиков, которые могли бы работать с кодом в дальнейшем.

Можно почитать про нейминг тут:
https://habr.com/ru/companies/dododev/articles/714512/

Nikitaaeee commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -6,62 +6,113 @@ import Foundation
+                } else {
+                    return "Рейтинг этого фильма меньше чем \(Int(randomOffset))?"
+                }
+            } ()
```
🔥Задание со звездочкой выполнено!

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -158,4 +163,38 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
+        } else {
+            activityIndicator.stopAnimating()
+        }
+    }
```
Входные параметры методов принято писать с маленькой буквы в camelCase, кроме того я бы рекомендовал заменить название параметра с `Bool` на что-то более конкретное и описательное, например, `shouldShow`. 
Использование `Bool` в качестве имени переменной может вводить в заблуждение, потому что `Bool` - это на самом деле название типа данных в Swift.

Кроме того, на первый взгляд, прочитав названия метода `showLoadingIndicator` можно предположить, что при входном параметре `true` наш индикатор должен появляться, а в методе все происходит наоборот. Можно попробовать переделать логику метода так, чтобы он соответствовал неймингу.

Есть крутой гайд по кодстайлу Swift от Ray Wenderlich, может быть полезным:
https://github.com/RedMadRobot/RMR-swift-style-guide#naming

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -158,4 +163,38 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
+        showLoadingIndicator(Bool: true)
+        questionFactory?.requestNextQuestion()
+    }
+    ///Метод при ошибки загрузки данных
```
Комментарии могут быть очень полезными, и я заметил, что ты прилагаешь усилия к тому, чтобы добавлять пояснения к некоторым методам/свойствам.

Однако, я хотел бы предложить более умеренный подход к комментированию. Хорошо структурированный и читаемый код часто говорит сам за себя. Если название метода или свойства ясно отражает его назначение и его поведение легко понять из контекста, то комментарий может быть не нужен.
Эта тема хорошо разобрана в книге "Чистый код" Роберта Мартина.
Также про комментарии в коде можно посмотреть тут:
https://habr.com/ru/companies/enterra/articles/182574/

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -70,22 +74,21 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
     /// Метод для отображения вопросов
-    private func showQuiz(quiz step: QuizStepViewModel) {
-        counterLabel.text = step.questionNumber
+    private func showQuiz(quiz step: QuizStepViewModel) {counterLabel.text = step.questionNumber
```
Тут код `counterLabel.text = step.questionNumber` можно перенести на следущую строку.

Nikitaaeee commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -6,62 +6,113 @@ import Foundation
+                imageData = try Data(contentsOf: movie.resizedImageURL)
+            } catch {
+                DispatchQueue.main.async {
+                    [weak self ] in guard let self = self else { return }
+                    self.delegate?.didFailToLoadData(with: NetworkClient.NetworkError.codeError)
```
Обработать ошибку загрузки изображения - отличная идея!

ASevastyanov committed:
Тех_долг
Доработка комментариев по ДЗ
Доработка логики вопроса/проверка правильности ответа

ASevastyanov committed:
Merge branch 'project_sprint_3_start' into sprint_06
